src/PresentationalComponents/TagsToolbar/TagsList.js

```diff
@@ -0,0 +1,85 @@
+import './_TagsToolbar.scss';
+
+import * as AppActions from '../../AppActions';
+
+import { Button } from '@patternfly/react-core/dist/js/components/Button/Button';
+import { Chip } from '@patternfly/react-core/dist/js/components/ChipGroup/Chip';
+import { ChipGroup } from '@patternfly/react-core/dist/js/components/ChipGroup/ChipGroup';
+import { DataList } from '@patternfly/react-core/dist/js/components/DataList/DataList';
+import { DataListCell } from '@patternfly/react-core/dist/js/components/DataList/DataListCell';
+import { DataListCheck } from '@patternfly/react-core/dist/js/components/DataList/DataListCheck';
+import { DataListItem } from '@patternfly/react-core/dist/js/components/DataList/DataListItem';
+import { DataListItemCells } from '@patternfly/react-core/dist/js/components/DataList/DataListItemCells';
+import { DataListItemRow } from '@patternfly/react-core/dist/js/components/DataList/DataListItemRow';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+import messages from '../../Messages';
+
+const TagsList = ({ setSelectedTags, selectedTags, showMoreCount, intl, tags, handleModalToggle }) => {
+    const updateSelectedTags = (value, e) => selectedTags.indexOf(e.target.name) > -1 ?
+        setSelectedTags(selectedTags.filter(item => item !== e.target.name))
+        : setSelectedTags([...selectedTags, e.target.name]);
+
+    return <React.Fragment>
+        {!showMoreCount && <ChipGroup>
+            {selectedTags.map(currentChip => (
+                <Chip key={currentChip} onClick={() => updateSelectedTags(null, { target: { name: currentChip } })}>
+                    {currentChip}
+                </Chip>
+            ))}
+        </ChipGroup>}
+        <div className='tagsListContainer'>
+            <DataList aria-label="tag-list">
+                {tags.slice(0, showMoreCount || tags.length).map(item => <DataListItem aria-labelledby="tag-list-item" key={item}>
```

Comment:
Just a tip for a some future work here. I would use extract the list item component to stand aline component and use `React.memo` around it to prevent unnecessary re-renders. :wink:

Reply:
👍 hokie dokie! there's a few places this oughta be done, we'll save that block of work to be done altogether 🙇‍♀